index.js

```diff
@@ -103,8 +104,9 @@ export default class Carousel extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.children !== nextProps.children) {
+    if (!isEqual(this.props.children, nextProps.children)) {
       let childrenLength = 0;
+      this.setState({ currentPage: 1 });
```

Comment:
When you

- mount carousel with some children
- swipe it to e.g. the 4th page
- replace the children programatically with a new set of only 2 pages

The carousel now displays the first page, but the info bullet says `4/2`.